ota_pal.c

```diff
@@ -73,14 +94,166 @@ static psa_image_id_t xOTAImageID = TFM_FWU_INVALID_IMAGE_ID;
 /* The key handle for OTA image verification. The key should be provisioned
  * before starting an OTA process by the user.
  */
-extern psa_key_handle_t xOTACodeVerifyKeyHandle;
+static psa_key_handle_t xOTACodeVerifyKeyHandle = ( psa_key_handle_t ) OTA_SIGNING_KEY_ID;
+
+static uint8_t ucECDSARAWSignature[ ECDSA_SHA256_RAW_SIGNATURE_LENGTH ] = { 0 };
+
+const AppVersion32_t appFirmwareVersion = { 0 };
 
 /***********************************************************************
  *
  * Functions
  *
  **********************************************************************/
 
+static bool prvConvertToRawECDSASignature( const uint8_t * pucEncodedSignature,  uint8_t * pucRawSignature )
+{
+    bool xReturn = true;
+    const uint8_t * pxNextLength = NULL;
+    uint8_t ucSigComponentLength;
+
+    if( ( pucRawSignature == NULL ) || ( pucEncodedSignature == NULL ) )
+    {
+        xReturn = false;
+    }
+
+    if( xReturn == true )
+    {
+        /* The signature has the format
+         * SEQUENCE LENGTH (of entire rest of signature)
+         *      INTEGER LENGTH  (of R component)
+         *      INTEGER LENGTH  (of S component)
```

Comment:
Can you clarify in the comments which standard this format comes from?

Reply:
It's ASN1-DER format. Added in comments.